### PR TITLE
Test input element directionality is only submitted for text and search type

### PR DESCRIPTION
**Pull request by vinhill**
This is in reference to html issue [9206](https://github.com/whatwg/html/issues/9206)

The standard currently does not enforce that for input elements, the directionality only applies for text and search types. I currently work on changing this and this pull request provides a test for the coming change.

### CODE REVIEW

**Comment by vinhill** [comment]
See pull request for WHATWG HTML [9207](https://github.com/whatwg/html/pull/9207)

**Review by annevk — COMMENTED** [review]
Add coverage for `textarea` too?

Stretch goal: might also want to add coverage for `new FormData(form)`.

Stretch goal 2: factor out the shared bits of scripts in this directory into a `resources/dirname.js` or some such and move `dirname-ltr-iframe.html` into that directory too.

**Comment by vinhill** [comment]
Made the changes and simplified the shared parts a bit, I hope the commit is not too large.

In the future, we could also test the textarea element in all other dirname test files. My suggestion is that I'll submit those changes when I get to implementing dirname on textarea in firefox.

File: html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+function onIframeLoaded(cb, selector="iframe") {
```
Comment (annevk) [inline html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js:1]: This function should take _t_ as parameter so the `load` listener below can be wrapped in `t.step_func_done`. And then `cb` won't have to be wrapped cause it runs synchronously as part of that.

File: html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  const iframe = document.querySelector(selector);
+  iframe.addEventListener("load", function() {
+    // The initial about:blank load event can be fired before the form navigation occurs.
+    // See https://github.com/whatwg/html/issues/490 for more information.
+    if(iframe.contentWindow.location.href == "about:blank") { return; }
+
+    const params = new URLSearchParams(iframe.contentWindow.location.search);
+    cb(params);
+  });
+}
```
Comment (annevk) [inline html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js:11]: Please add a trailing newline.

**Review by annevk — COMMENTED** [review]
This looks great, happy to land this with some minor nits addressed.

File: html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js (hunk 1 above)
Comment (vinhill) [reply html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js:1]: That would not work in `dirname-only-if-applies.html`, as I'm not using an `async_test` there. I wanted to have separate subtests for `input` and `textarea` and from my understanding, I cannot have two `async_test`s at the same time.

Comment (annevk) [reply html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js:1]: I see, but that is a bit messy mixing both an explicit_done and having several nested tests.

That makes it very easy to create a harness failure which would result in the additional tests not getting registered.

Comment (vinhill) [reply html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js]: Okay, so I applied the change and now use two asynchronous tests.
I believe, it's better then to have done in the function name. Also, using `step` only on `cb` leaves the function less nested.

File: html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    const params = new URLSearchParams(iframe.contentWindow.location.search);
+    t.step(() => cb(params))
+    if (done) t.done();
+  });
+}
+
+function onIframeLoaded(t, cb, selector="iframe") {
+  return __onIframeLoaded(t, cb, selector, false);
+}
```
Comment (annevk) [inline html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js:16]: Am I blind or are there no callers for this algorithm?

**Review by annevk — COMMENTED** [review]
Thanks for cleaning this up and sorry for the delay.

File: html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js (hunk 3 above)
Comment (vinhill) [reply html/semantics/forms/attributes-common-to-form-controls/resources/dirname.js:16]: Yeah, I wasn't sure if I needed this later on, so I included both versions. Do you want me to remove it?

**Review by annevk — APPROVED** [review]

**Pull request merged**